node/src/runner.rs

```diff
@@ -71,11 +71,14 @@ pub async fn initialize(
     );
 
     // P2P subsystem
+    let p2p_config = node_config.p2p.into();
+    let p2p_transport = p2p::net::libp2p::make_transport(&p2p_config);
```

Comment:
This function should be moved to `P2p::new()`.

The idea is that depending on the backend chosen (with the generic parameter of `make_p2p()`), the correct backend with the correct transport will be created. There's no need to parametrize transport all the way to the node executable.

Reply:
Done, thanks.